app/src/services/kleros.ts

```diff
@@ -0,0 +1,133 @@
+import { Contract, ethers } from 'ethers'
+
+import { Token } from '../util/types'
+import { getContractAddress, getTokensByNetwork, networkIds } from '../util/networks'
+import { Web3Provider } from 'ethers/providers'
+
+const klerosBadgeAbi = [
+  'function queryAddresses(address _cursor, uint _count, bool[8] _filter, bool _oldestFirst) external view returns (address[] values, bool hasMore)',
+]
+
+const klerosTokensViewAbi = [
+  'function getTokensIDsForAddresses(address _t2crAddress, address[] _tokenAddresses ) external view returns (bytes32[] result)',
+  'function getTokens(address _t2crAddress, bytes32[] _tokenIDs ) external view returns (tuple(bytes32 ID, string name, string ticker, address addr, string symbolMultihash, uint8 status, uint256 decimals)[] result)',
```

Comment:
Check this, how to manage a struct (tuple) with human readable abi

Reply:
Well, if it works, it works.